### PR TITLE
Replace simple [[ ... ]] conditions with [ ... ]

### PR DESCRIPTION
**Pull request by y0rune**
Hello!
I used the `sd` for issue (https://github.com/webinstall/webi-installers/issues/465) 

```
# Example:
# Before: [[ cos || cos ]]
# After: [ cos ] || [ cos ]
sd '\[\[ (.*) \|\| (.*) \]\]' '[ $1 ] || [ $2 ]' *.sh */*.sh */*/*.sh

# Example:
# Before: [[ ! cos ]]
# After: ! [ cos ]
sd '\[\[ ! (.*) \]\]' '! [ $1 ]' *.sh */*.sh */*/*.sh

# Remove another cases
sd '\[\[' '[' */*.sh
sd '\]\]' ']' */*.sh
```

One line command:
```sh
sd '\[\[ (.*) \|\| (.*) \]\]' '[ $1 ] || [ $2 ]' *.sh */*.sh */*/*.sh ; sd '\[\[! (.*) \]\]' '! [ $1 ]' *.sh */*.sh */*/*.sh ; sd '\[\[' '[' */*.sh ; sd '\]\]' ']' */*.sh
```

Thank you!

### CODE REVIEW

**Comment by coolaj86** [comment]
This isn't something that can just be done automatically. It needs careful inspection of each case to make sure that it doesn't change the behavior.

POSIX `[` does not behave like Bash `[[`.

**Comment by y0rune** [comment]
All right I test it all ;)

**Comment by y0rune** [comment]
List of tested scripts:
- [ ] _webi/bootstrap.sh
- [ ] _webi/template.sh
- [ ] brew/install.sh
- [ ] git/install.sh
- [ ] gpg-pubkey/gpg-pubkey.sh
- [ ] gpg/install.sh
- [ ] iterm2/install.sh
- [ ] myip/myip.sh
- [ ] python/install.sh
- [ ] python2/install.sh
- [ ] ssh-adduser/ssh-adduser.sh
- [ ] yq/install.sh

**Comment by coolaj86** [comment]
Thanks. I'm going to hold this one off a little longer until we're ready to switch `/bin/bash` to `/bin/sh`.

I'll try to get the others merged in today.

**Comment by y0rune** [comment]
All right :+1:

**Comment by coolaj86** [comment]
I found a solution to another of the `[[` bashisms:

```diff
- if [[ "$-" = *i* ]]; then
- if echo "$-" | grep -q 'i'; then
```

See <https://stackoverflow.com/questions/73412543/how-to-match-a-substring-using-posix-shell>.

File: _webi/bootstrap.sh [hunk 1]
```diff
@@ -182,12 +182,12 @@ function __webi_main () {
         echo ""
     }
 
-    if [[ \$# -eq 0 ]] || [[ "\$1" =~ ^(-V|--version|version)$ ]]; then
+    if [ \$# -eq 0 ] || [ "\$1" =~ ^(-V|--version|version)$ ]; then
```
Comment (coolaj86) [inline _webi/bootstrap.sh:185]: I don't believe that =~ is available in Posix.

Probably something like `echo "\$1" | grep -q '^(-V|--version|version)$'`

File: _webi/template.sh [hunk 2]
```diff
@@ -181,7 +181,7 @@ function __bootstrap_webi() {
             # TODO wget -c --content-disposition "$my_url"
             set +e
             my_show_progress=""
-            if [[ $- == *i* ]]; then
+            if [ $- == *i* ]; then
```
Comment (coolaj86) [inline _webi/template.sh:184]: `if echo "$-" | grep -q 'i';` should work here

File: brew/install.sh [hunk 3]
```diff
@@ -7,9 +7,9 @@ function _install_brew() {
     # Straight from https://brew.sh
     #/bin/bash -c "$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/master/install.sh)"
 
-    if [[ -n "$(uname -a | grep -i darwin)" ]]; then
+    if [ -n "$(uname -a | grep -i darwin)" ]; then
```
Comment (coolaj86) [inline brew/install.sh:10]: I think this should be just `if uname -a | grep -q -i darwin;`

File: git/install.sh [hunk 4]
```diff
@@ -5,7 +5,7 @@ set -u
 function __init_git() {
 
     if [ -z "$(command -v git)" ]; then
-        if [[ -n "$(uname -a | grep -i darwin)" ]]; then
+        if [ -n "$(uname -a | grep -i darwin)" ]; then
```
Comment (coolaj86) [inline git/install.sh:8]: same here as above

File: ssh-adduser/ssh-adduser.sh [hunk 5]
```diff
@@ -56,20 +56,20 @@ function main() {
     # TODO ensure that ssh-password login is off
     my_pass="$(grep 'PasswordAuthentication yes' /etc/ssh/sshd_config)"
     my_pam=""
-    if [[ "Darwin" == "$(uname -s)" ]]; then
+    if [ "Darwin" == "$(uname -s)" ]; then
```
Comment (coolaj86) [inline ssh-adduser/ssh-adduser.sh:59]: This needs to be `[ "Darwin" = "$(uname -s)" ]`

**Comment by coolaj86** [comment]
`[` is actually a command, located at `/bin/[`. It is an alias of `/bin/test`. When called as `[` then the last string passed to it must be `]`.

This is the Posix standard for `test` (a.k.a. `[`): https://pubs.opengroup.org/onlinepubs/9699919799/utilities/test.html

Please take a look and see if you notice any non-standard uses that I missed.

**Comment by y0rune** [comment]
All right I will work on it on Monday ;)

File: _webi/bootstrap.sh [hunk 6]
```diff
@@ -182,12 +182,12 @@ function __webi_main () {
...
         version
         exit 0
     fi
 
-    if [[ "\$1" =~ ^(-h|--help|help)$ ]]; then
+    if [ "\$1" =~ ^(-h|--help|help)$ ]; then
```
Comment (coolaj86) [inline _webi/bootstrap.sh:190]:
```sh
if echo "\$1" | grep -q -E '^(-h|--help|help)$'; then
```

File: _webi/template.sh (hunk 2 above)
Comment (coolaj86) [reply _webi/template.sh:184]: I did find that there's a "simpler" way to do this (using only Posix Shell, not grep), but it's not as easy to read:
https://stackoverflow.com/a/73419937/151312

**Comment by coolaj86** [comment]
Some of the other replacements that will need to come in a follow-up...

```sh
if echo "\$1" | grep -q -E '^(-h|--help|help)$'; then
if echo "$-" | grep -q 'i'; then
if uname -a | grep -q -i darwin; then
```

**Review by coolaj86 — APPROVED** [review]
Needed some automatable double-quotes added, and the `[[ = > [` that need complex changes reverted

**Comment by coolaj86** [comment]
@y0rune Please open a follow-up PR for the remaining changes. \
(mentioned above)

**Pull request merged**